src/main/java/com/github/princesslana/eriscasper/util/Shard.java

```diff
@@ -0,0 +1,49 @@
+package com.github.princesslana.eriscasper.util;
+
+import com.github.princesslana.eriscasper.ErisCasperFatalException;
+import com.github.princesslana.eriscasper.data.immutable.Wrapper;
+import com.ufoscout.properlty.Properlty;
+import java.util.Optional;
+
+public class Shard implements Wrapper<Integer[]> {
+
+  private final int shardNumber;
+  private final int shardTotal;
+
+  public Shard(int shardNumber, int shardTotal) {
+    if (shardNumber >= shardTotal || shardTotal < 1 || shardNumber < 0) {
```

Comment:
`Preconditions.checkArgument` from Guava works well for this kinds of checks. It'll throw `IllegalArgumentException` instead of `ErisCasperFatalException`. I can see arguments for either.

Reply:
I don't think it's necessary to mix exceptions. It's also a very primitive thing to check, I don't think it's necessary to wrap it in Preconditions.

Reply:
The judgement should be made based on readability. I agree it's not a big deal with exception is throw and that the code is straight forward. Either changing or leaving is fine.

```
Preconditions.checkArgument(shardNumber >= 0, "shardNumber should be zero or more");
Preconditions.checkArgument(shardTotal >= 1, "shardTotal should be one or more");
Preconditions.checkState(shardNumber < shardTotal, "shardNumber should be less than shardTotal");
```

Reply:
I might change it to preconditions then I do think that is more descriptive, tests will change with it